examples/jetstream/queue-push-consumer/go/main.go

```diff
@@ -0,0 +1,159 @@
+	// and group name.
+	sub2, _ := js.QueueSubscribeSync("events.>", "event-processor", nats.AckExplicit())
+
+	// As noted above, a queue push consumer relies on a core NATS
+	// queue group for distributing messages to active members. As
+	// a result, we can bind a subsription by using the `DeliverSubject`
+	// and the `DeliverGroup`
+	// Since messages are publishing to a dedicated subject and is
+	// part of a group, we can also create a core NATS subscription
+	// to join the group. As a resminder, the `DeliverSubject` is
+	// randomly generated by default, but this can be set explicitly
+	// in the consumer config if desired.
+	sub3, _ := nc.QueueSubscribeSync(info.Config.DeliverSubject, info.Config.DeliverGroup)
+	fmt.Printf("deliver subject: %q\n", info.Config.DeliverSubject)
+
+	// Now we can publish some messages to the stream to observe how
+	// they are distributed to the subscribers.
+	js.Publish("events.1", nil)
+	js.Publish("events.2", nil)
+	js.Publish("events.3", nil)
+	js.Publish("events.4", nil)
+	js.Publish("events.5", nil)
+	js.Publish("events.6", nil)
+
+	// As noted in the [push consumer][1] example, subscriptions
+	// enqueue messages proactively. When there are a group of
+	// subscriptions, each will receive a different subset of the messages.
+	// When calling `NextMsg` this means, messages can be processed out
+	// of order. There is no correlation with message order and
+	// subscription creation order 😉. In fact, it is possible that
+	// not all subscriptions will necessarily get a message.
+	// [1]: /examples/jetstream/push-consumer/go/
+	msg, _ := sub1.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub1: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub1: receive timeout")
+	}
+
+	msg, _ = sub2.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub2: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub2: receive timeout")
+	}
+
+	msg, _ = sub3.NextMsg(time.Second)
+	if msg != nil {
+		fmt.Printf("sub3: received message %q\n", msg.Subject)
+		msg.Ack()
+	} else {
+		fmt.Println("sub3: receive timeout")
+	}
+
+	// Like all ephemeral consumers, once the subscription that created
+	// it unsubscribes (or calls `Drain`), the consumer will be deleted.
+	sub1.Unsubscribe()
+	sub2.Unsubscribe()
+	sub3.Unsubscribe()
+
+	// ### Durable
+	// To create a (safe) durable consumer, use the `AddConsumer`
+	// method. Although it may seem redundant, a durable name *and*
+	// the deliver group name must be defined. This is simply because
+	// the durable name is used for all consumer types, while the
+	// deliver group is exclusive to the queue push consumer. In this
+	// example, the same name is used as convention.
+	fmt.Println("\n# Durable")
+
+	js.AddConsumer(streamName, &nats.ConsumerConfig{
+		Durable:        "event-processor",
+		DeliverSubject: "my-subject",
+		DeliverGroup:   "event-processor",
+		AckPolicy:      nats.AckExplicitPolicy,
+	})
```

Comment:
The main difference between creating via js.QueueSubscribe and js.AddConsumer I think is that when auto-creating with QueueSubscribe it will also delete the consumer when calling either Drain or Unsubscribe (something that we need to revise...)
When using AddConsumer to create and then bind with QueueSubscribe, that does not happen so does become safer for the subscribers to call `sub.Drain` since the consumer will not be cleaned up.